File: php/exercises/2.gymRegister/php/plans.php
<?php

trait plans {

private string $mensageError="";

public function plan(){
    
    if($this->getPlan()=="basic") {
        echo "<h1>THE PLAN SELECT IS: BASIC</h1>";
        return "This plan have a cost  of $".  34000 . " access general of rhe GYM";
    }else 
    if($this->getPlan()=="medium"){
        echo "<h1>THE PLAN SELECT IS: MEDIUM</h1>";
        return "This plan have a cost  of $". 56000 . " includes directed classes in the GYM ";
    }else
    if($this->getPlan()=="premium"){
        echo "<h1>THE PLAN SELECT IS: PREMIUM</h1>";
        return  "This plan have a cost  of $".   88000 ." Includes personal trainer and wet area in the GYM";
    }else{  echo "<h1>THE PLAN SELECT IS: ???</h1>";
            echo $this->mensageError="error this isn't a plan";
    }
}

    public function showPlan(){
        echo $this->mensageError;
        return $this->plan();  
    }
}
?>

File: java/GymRegistrer/src/main/java/com/maicol/gymregistrer/Plans.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.gymregistrer;

/**
 *
 * @author Maicol
 */
public class Plans {
    private String errorMessage = "";

    public String plan(String planName) {
        switch (planName) {
            case "basic" -> {
                System.out.println("<h1>THE PLAN SELECT IS: BASIC</h1>");
                return "This plan have a cost  of $34000 access general of the GYM";
            }
            case "medium" -> {
                System.out.println("<h1>THE PLAN SELECT IS: MEDIUM</h1>");
                return "This plan have a cost  of $56000 includes directed classes in the GYM";
            }
            case "premium" -> {
                System.out.println("<h1>THE PLAN SELECT IS: PREMIUM</h1>");
                return "This plan have a cost  of $88000 Includes personal trainer and wet area in the GYM";
            }
            default -> {
                System.out.println("<h1>THE PLAN SELECT IS: ???</h1>");
                errorMessage = "error this isn't a plan";
                return errorMessage;
            }
        }
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}